src/NuGet.Clients/PackageManagement.UI/Models/PackageItemListViewModel.cs

```diff
@@ -265,6 +266,159 @@ internal Lazy<Task<BackgroundLoaderResult>> BackgroundLoader
 
         public Uri IconUrl { get; set; }
 
+        // same URIs can reuse the bitmapImage that we've already used.
+        static Dictionary<Uri, BitmapImage> bitmapImageCache = new Dictionary<Uri, BitmapImage>();
```

Comment:
The images are still in the internet cache, but I didn't think that the BitmapImages needed to be kept around.
We could also chose to clear the cache when new search terms are made.
Should likely be done in parallel with your caching of the models for the packages...I didn't investigate the model side of your house.